src/utils/getUserRewards.ts

```diff
@@ -160,21 +160,23 @@ const computeIndex = async (
   }, lastIndex);
 };
 
-export const getUserBalances = async (graphUrl: string, user: string, block?: number) =>
-  axios
-    .post<
-      { query: string; variables: { user: string } },
-      { data: { data?: { user?: GraphUserBalances }; errors?: any } }
-    >(graphUrl, {
+export const getUserBalances = async (graphUrl: string, user: string, block?: number) => {
+  const res = await fetch(graphUrl, {
+    method: "POST",
+    headers: { "Content-Type": "application/json" },
+    body: JSON.stringify({
       query: block ? queryWithBlock : query,
       variables: { user, block },
-    })
-    .then((r) => {
-      if (!r.data?.data) throw Error(JSON.stringify(r.data.errors));
-      if (!r.data.data.user) return undefined;
-      return formatGraphBalances(r.data.data.user);
-    });
+    }),
+  });
+  if (!res.ok) throw Error(res.status.toString());
```

Comment:
What if there is a non 200 resp but with more information in the body? here for a subgraph error, we have more information in `res.data.errors` for example